src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php

```diff
@@ -59,6 +59,24 @@ class AttributeMetadata implements AttributeMetadataInterface
      */
     public $ignore = false;
 
+    /**
+     * @var array[] Normalization contexts per group name (or empty string for all groups)
+     *
+     * @internal This property is public in order to reduce the size of the
+     *           class' serialized representation. Do not access it. Use
+     *           {@link getNormalizationContexts()} instead.
+     */
+    public $normalizationContexts = [];
+
+    /**
+     * @var array[] Denormalization contexts per group name ("*" applies to all groups)
```

Comment:
Using `*` should be fine since https://github.com/symfony/symfony/pull/33540 already relies on this. So no one should name their groups `*`.
Otherwise, previous version used an empty string, but we could simply opt for a distinct `defaultDenormalizationContext` as well.